rbac/api/cross_access/view.py

```diff
@@ -260,3 +258,60 @@ def update_status(self, car, status):
+        if request.user.account == update_obj.target_account:
+            """ For approvers updating requests coming to them, only org admin
+                could update status from pending/approved/denied to approved/denied.
+            """
+            if not request.user.admin:
+                self.throw_validation_error("cross-account partial update", "Only org admin could update status.")
+            if update_obj.status not in ["pending", "approved", "denied"]:
+                self.throw_validation_error(
+                    "cross-account partial update", "Only pending/approved/denied requests could be updated."
+                )
+            if request.data.get("status") not in ["approved", "denied"]:
+                self.throw_validation_error(
+                    "cross-accont partial update", f"Status could only be updated to approved/denied."
+                )
+            if len(request.data.keys()) > 1 or next(iter(request.data)) != "status":
+                self.throw_validation_error("cross-accont partial update", f"Only status could be updated.")
+        elif request.user.user_id == update_obj.user_id:
+            """ For requestors updating their requests, the status update could
+                only be from pending to cancelled.
+            """
+            if update_obj.status != "pending" or request.data.get("status") != "cancelled":
+                self.throw_validation_error(
+                    "cross-accont partial update", f"Status could only be updated from pending to cancelled."
+                )
+            for field in request.data:
+                if field not in VALID_PATCH_FIELDS:
+                    self.throw_validation_error(
+                        "cross-accont partial update",
+                        f"Field '{field}' is not supported. Please use one or more of: {VALID_PATCH_FIELDS}",
+                    )
+        else:
+            self.throw_validation_error(
+                "cross-accont partial update",
+                f"User does not have permission to update the request that does not belong to them.",
+            )
+
+    def check_update_permission(self, request, update_obj):
+        """Check if user has permission to update cross access request."""
+        # Only requestors could update the cross access request.
+        if request.user.user_id != update_obj.user_id:
+            self.throw_validation_error(
+                "cross-account update", "Only requestor could update the cross access request."
+            )
+
+        # Only pending request could be updated.
+        if update_obj.status != "pending":
+            self.throw_validation_error("cross-account update", "Only pending requests could be updated.")
+
+        # Do not allow updating the status:
+        if request.data.get("status") and str(request.data.get("status")) != "pending":
+            self.throw_validation_error("cross-account update", "Please use partial update to update the status.")
```

Comment:
This message is a bit confusing. Can we be more explicit in it for what the client would need to do?

Reply:
Sure, will try to add more context